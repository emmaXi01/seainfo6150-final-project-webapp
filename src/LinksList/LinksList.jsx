import React from "react";
import PropTypes from "prop-types";
import styles from "./LinksList.module.css";

const LinkList = (props) => {
  const displayContent = (
    <ul className={styles.linksList}>
        { props.links.map((link) => <li key={link}>{link}</li>) }
    </ul>
  );

  return (
    <div>
      { displayContent }
    </div>
  );
};

LinkList.propTypes = {
  links: PropTypes.array.isRequired
}

export default LinkList;