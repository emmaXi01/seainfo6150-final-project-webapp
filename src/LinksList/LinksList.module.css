.linksList {
  list-style-type: none;
  font-size: .875rem;
  padding-left: 0px;
}