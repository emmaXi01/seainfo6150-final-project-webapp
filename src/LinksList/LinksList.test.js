import React from "react";
import { render } from "@testing-library/react";
import LinksList from "./LinksList.jsx";

describe("LinksList tests", () => {
  it("renders the LinksList component correctly", () => {
    const links= ["About", "Leadership", "Careers", "Catalog"];  
    const { container } = render(<LinksList links={links} />);
    expect(container).toMatchSnapshot();
  });
});