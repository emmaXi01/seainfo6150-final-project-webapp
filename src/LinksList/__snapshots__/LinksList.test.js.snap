// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinksList tests renders the LinksList component correctly 1`] = `
<div>
  <div>
    <ul
      class="linksList"
    >
      <li>
        About
      </li>
      <li>
        Leadership
      </li>
      <li>
        Careers
      </li>
      <li>
        Catalog
      </li>
    </ul>
  </div>
</div>
`;
